Frameworks/CoreGraphics/CGContextCairo.mm

```diff
@@ -1977,3 +1985,62 @@
     cairo_path_destroy(caPath);
     return (CGPathRef)copyPath;
 }
+
+/**
+ * Helper method to render text with the given DWRITE_GLYPH_RUN.
+ * Today we create a IWICBitmap with the underlying XAML Writable Bitmap raw bytes and rasterize to it before copying it back to the
+ * original buffer.
+ * TODO::
+ *    1. This method today does not handle DPI scaling. Investigate what needs to be one here to support it
+ *    2. There is double copy happening here - first when the IWICBitmap is created and later the rasterized image is copied back
+ *       Implementing our own IWICBitmap interface should fix this issue.
+ *
+ * @parameter glyphRun DWRITE_GLYPH_RUN object to render
+ */
+void CGContextCairo::CGContextDrawGlyphRun(const DWRITE_GLYPH_RUN* glyphRun) {
+    ObtainLock();
+    LOCK_CAIRO();
+
+    CGContextStrokePath();
+
+    BYTE* buffer = static_cast<BYTE*>(_imgDest->Backing()->LockImageData());
+    THROW_NS_IF_NULL(E_UNEXPECTED, buffer);
+    int stride = _imgDest->Backing()->BytesPerRow();
+    float width = _imgDest->Backing()->Width();
+    float height = _imgDest->Backing()->Height();
+
+    ComPtr<IWICImagingFactory> wicFactory;
+    CoCreateInstance(CLSID_WICImagingFactory, NULL, CLSCTX_INPROC_SERVER, __uuidof(IWICImagingFactory), &wicFactory);
+    THROW_NS_IF_NULL(E_UNEXPECTED, wicFactory);
+
+    ComPtr<IWICBitmap> wicBitmap;
+    THROW_IF_FAILED(
+        wicFactory->CreateBitmapFromMemory(width, height, GUID_WICPixelFormat32bppPBGRA, stride, height * stride, buffer, &wicBitmap));
+    THROW_NS_IF_NULL(E_UNEXPECTED, wicBitmap);
```

Comment:
I guess I got the wrong file version checked in. I should cover CoCreateInstance with THROW_IF_FAILED

---
In reply to: [78452483](https://github.com/Microsoft/WinObjC/pull/999#discussion_r78452483) [](ancestors = 78452483)